spring-integration-gcp/src/test/java/org/springframework/integration/gcp/storage/inbound/GCSInboundFileSynchronizerTest.java

```diff
@@ -0,0 +1,140 @@
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.BDDMockito;
+
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.integration.config.EnableIntegration;
+import org.springframework.integration.file.filters.AcceptOnceFileListFilter;
+import org.springframework.messaging.Message;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+
+/**
+ * @author João André Martins
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration
+public class GCSInboundFileSynchronizerTest {
+
+	@Autowired
+	private Storage gcs;
+
+	private static final Log LOGGER = LogFactory.getLog(GCSInboundFileSynchronizerTest.class);
+
+	@After
+	@Before
+	public void cleanUp() throws IOException {
+		Path testDirectory = Paths.get("test");
+
+		if (Files.exists(testDirectory)) {
+			if (Files.isDirectory(testDirectory)) {
+				Files.list(testDirectory).forEach(path -> {
+					try {
+						Files.delete(path);
+					}
+					catch (IOException ioe) {
+						LOGGER.info("Error deleting test file.", ioe);
+					}
+				});
+			}
+
+			Files.delete(testDirectory);
+		}
+	}
+
+	@Test
+	public void testCopyFiles() throws Exception {
+		File localDirectory = new File("test");
+		GCSInboundFileSynchronizer synchronizer = new GCSInboundFileSynchronizer(this.gcs);
+		synchronizer.setRemoteDirectory("test-bucket");
+		synchronizer.setBeanFactory(mock(BeanFactory.class));
+
+		GCSInboundFileSynchronizingMessageSource adapter = new GCSInboundFileSynchronizingMessageSource(synchronizer);
+		adapter.setAutoCreateLocalDirectory(true);
+		adapter.setLocalDirectory(localDirectory);
+		adapter.setBeanFactory(mock(BeanFactory.class));
+
+		adapter.setLocalFilter(new AcceptOnceFileListFilter<>());
+
+		adapter.afterPropertiesSet();
+
+		Message<File> message = adapter.receive();
+		assertThat(message.getPayload().getName()).isEqualTo("legend of heroes");
+		assertThat(Files.readAllBytes(message.getPayload().toPath())).isEqualTo("estelle".getBytes());
+
+		message = adapter.receive();
+		assertThat(message.getPayload().getName()).isEqualTo("trails in the sky");
+		assertThat(Files.readAllBytes(message.getPayload().toPath())).isEqualTo("joshua".getBytes());
+
+		message = adapter.receive();
+		assertThat(message).isNull();
+	}
+
+	@Configuration
+	@EnableIntegration
+	public static class Config {
+
+		@Bean
+		public Storage gcs() {
+			Storage gcsMock = mock(Storage.class);
+
+			Blob blob1 = mock(Blob.class);
+			Blob blob2 = mock(Blob.class);
+
+			BDDMockito.willAnswer(invocation -> "legend of heroes").given(blob1).getName();
```

Comment:
This can be statically imported.
Yes, nit-pick, but again code style 😄

Reply:
I'll add this new `org.mockito.BDDMockito` package to the checkstyle rules.